File: developers/academy/py/standalone/which_search/_snippets/05_review.py
from uuid import UUID
import os

weaviate_url = os.getenv("WEAVIATE_URL")
weaviate_key = os.getenv("WEAVIATE_API_KEY")

# START connectionCode
import weaviate
import weaviate.classes as wvc

client = weaviate.connect_to_weaviate_cloud(
    cluster_url=weaviate_url,
    auth_credentials=wvc.init.Auth.api_key(weaviate_key),
    headers={
        "X-OpenAI-Api-Key": os.getenv("OPENAI_APIKEY") # Replace with your OpenAI API key (for vector and hybrid searches)
    }
)
# END connectionCode

# START nearTextExample
questions = client.collections.get("JeopardyQuestion")
response = questions.query.near_text(
    query="space travel",  # Your query string
    limit=2
)

for o in response.objects:
    print(o.uuid)
    print(o.properties)
# END nearTextExample

assert len(response.objects) == 2
assert "question" in response.objects[0].properties.keys()


response = questions.query.near_text(
    query="space travel",
    limit=1,
    include_vector=True
)
vector_input = response.objects[0].vector
object_input = response.objects[0].uuid

# START nearVectorExample
questions = client.collections.get("JeopardyQuestion")
response = questions.query.near_vector(
    near_vector=vector_input,  # Your vector object
    limit=2
)

for o in response.objects:
    print(o.uuid)
    print(o.properties)
# END nearVectorExample

assert len(response.objects) == 2
assert "question" in response.objects[0].properties.keys()


# START nearObjectExample
questions = client.collections.get("JeopardyQuestion")
response = questions.query.near_object(
    near_object=object_input,  # Your object UUID
    limit=2
)

for o in response.objects:
    print(o.uuid)
    print(o.properties)
# END nearObjectExample

assert len(response.objects) == 2
assert "question" in response.objects[0].properties.keys()


# START bm25Example
questions = client.collections.get("JeopardyQuestion")
response = questions.query.bm25(
    query="space travel",  # Your query string
    limit=2
)

for o in response.objects:
    print(o.uuid)
    print(o.properties)
# END bm25Example

assert len(response.objects) == 2
assert "question" in response.objects[0].properties.keys()


# START hybridExample
questions = client.collections.get("JeopardyQuestion")
response = questions.query.hybrid(
    query="space travel",  # Your query string
    limit=2
)

for o in response.objects:
    print(o.uuid)
    print(o.properties)
# END hybridExample

assert len(response.objects) == 2
assert "question" in response.objects[0].properties.keys()

File: developers/academy/js/standalone/which-search/_snippets/05_review.ts
import weaviate, { WeaviateClient, WeaviateReturn } from "weaviate-client"
// START connectionCode
let client: WeaviateClient


client = await weaviate.connectToWeaviateCloud(process.env.WCD_URL as string, {
  authCredentials: new weaviate.ApiKey(process.env.WCD_API_KEY as string),
  headers: {
    'X-OpenAI-Api-Key': process.env.OPENAI_APIKEY as string,  // Replace with your inference API key
  }
}
)
// END connectionCode


// START nearTextExample // START nearVectorExample // START nearObjectExample // START bm25Example // START hybridExample
type NonGenericReturn = WeaviateReturn<undefined>
let response: NonGenericReturn

const questions = client.collections.get("JeopardyQuestion")

// END nearTextExample // END nearVectorExample // END nearObjectExample // END bm25Example // END hybridExample

// START nearTextExample 
response = questions.query.nearText("space travel", // Your query string
  {
    limit: 2
  }
)

for (const item of response.objects) {
  console.log(item.uuid)
  console.log(item.properties)
}
// END nearTextExample
// assert len(response.objects) == 2
// assert "question" in response.objects[0].properties.keys()

response = questions.query.nearText("space travel", {
  limit: 1,
  includeVector: true
}
)
let vectorInput = response.objects[0].vectors
let objectInput = response.objects[0].uuid

// START nearVectorExample
response = questions.query.nearVector(vectorInput, // Your vectors
  {
    limit: 2
  }
)

for (const item of response.objects) {
  console.log(item.uuid)
  console.log(item.properties)
}
// END nearVectorExample

// assert len(response.objects) == 2
// assert "question" in response.objects[0].properties.keys()

// START nearObjectExample
response = questions.query.nearObject(objectInput, // Your object UUID
  {
    limit: 2
  }
)

for (const item of response.objects) {
  console.log(item.uuid)
  console.log(item.properties)
}
// END nearObjectExample


// assert len(response.objects) == 2
// assert "question" in response.objects[0].properties.keys()

// START bm25Example
response = questions.query.bm25("space travel", // Your query string
  {
    limit: 2
  }
)

for (const item of response.objects) {
  console.log(item.uuid)
  console.log(item.properties)
}
// END bm25Example

// assert len(response.objects) == 2
// assert "question" in response.objects[0].properties.keys()


// START hybridExample
response = questions.query.hybrid("space travel", // Your query string
  {
    limit: 2
  }
)

for (const item of response.objects) {
  console.log(item.uuid)
  console.log(item.properties)
}

// END hybridExample

// assert len(response.objects) == 2
// assert "question" in response.objects[0].properties.keys()
